diesel/src/query_source/joins.rs

```diff
@@ -292,6 +261,22 @@ impl<T, Join, On> AppearsInFromClause<T> for JoinOn<Join, On> where
     type Count = Join::Count;
 }
 
+#[allow(missing_debug_implementations, missing_copy_implementations)]
+#[doc(hidden)]
+/// A hack to allow bidirectional joins to be generated from `#[belongs_to]`
+///
+/// This type needs to exist because it is illegal in Rust today to write
```

Comment:
Yeah, that was why I haven't opened an issue. I figured the answer would either be "this is intentional" or "this is blocked on chalk". :D